### PR TITLE
Handle PCI paths consistently and more generally

### PR DESCRIPTION
**Pull request by dgibson**
This series is designed to change the agent to handle PCI paths more consistently, see issue #854 for more information.

### CODE REVIEW

File: device.go [hunk 1]
```diff
@@ -79,6 +79,17 @@ type devIndexEntry struct {
 }
 type devIndex map[string]devIndexEntry
 
+// Guest-side PCI path, identifies a PCI device by where it sits in
+// the PCI topology.
+//
+// Has the format "bridgeAddr/deviceAddr" where bridgeAddr is the
```
Comment (c3d) [inline device.go:85]: I thought from #854 that you could have multiple bridge addresses (device accessed with more than one bridge). So I think you see that as just a current limitation of the implementation. In that case, simply state "Currently has the format..." and then explain where this is going?

File: device.go [hunk 2]
```diff
@@ -93,15 +104,14 @@ func rescanPciBus() error {
...
-// while deviceAddr is the address at which the device is attached on the bridge.
-func getDevicePCIAddressImpl(pciID string) (string, error) {
-	tokens := strings.Split(pciID, "/")
+// pciPathToSysfs fetches the sysfs path for a PCI path, relative to
+// the syfs path for the PCI host bridge, based on the PCI path
+// provided.
+func pciPathToSysfsImpl(pciPath PciPath) (string, error) {
+	tokens := strings.Split(pciPath.path, "/")
 
 	if len(tokens) != 2 {
```
Comment (c3d) [inline device.go:113]: In the future, that test would be reworked, right?

**Review by c3d — APPROVED** [review]
Looks good to me. Two questions, just to educate myself.

File: device.go (hunk 1 above)
Comment (dgibson) [reply device.go:85]: Well, I figure that will become obvious when I add the patches to actually do that.  It's `do-not-merge` for a reason..

File: device.go (hunk 2 above)
Comment (dgibson) [reply device.go:113]: Yes.

**Comment by devimc** [comment]
thanks @dgibson  - lgtm but travis is failing 

```
Found 3 commits between commit f2db939b620618987d350f941a0c8d13941c7db1 and branch master

ERROR: No "Fixes" found

ERROR: checkcommits failed. See the document below for help on formatting

commits for the project.
```

**Comment by devimc** [comment]
now a different error :nerd_face: 

```
/home/travis/gopath/src/github.com/kata-containers/agent/protocols/mockserver

device.go:45: File is not `gofmt`-ed with `-s` (gofmt)
	sysBusPrefix        = sysfsDir + "/bus/pci/devices"
	pciBusRescanFile    = sysfsDir + "/bus/pci/rescan"
	pciBusPathFormat    = "%s/%s/pci_bus/"
	systemDevPath       = "/dev"
	getSCSIDevPath      = getSCSIDevPathImpl
	getPmemDevPath      = getPmemDevPathImpl
	getPCIDeviceName    = getPCIDeviceNameImpl
	pciPathToSysfs      = pciPathToSysfsImpl
	scanSCSIBus         = scanSCSIBusImpl
```

**Comment by jodh-intel** [comment]
Restarted Debian CI which timed out.

Also restarted the CentOS one but I think we may have seen this issue before @GabyCT?

```
bats integration/netmon/netmon_test.bats
1..1
not ok 1 test netmon
# (in test file integration/netmon/netmon_test.bats, line 60)
#   `[ "$after_interfaces" -gt "$before_interfaces" ]' failed
# 
# e6153b744972f86bffc5573fdc0f2a52eea49ebdcb78510d3599aa5e13ffcfb6
# 0ee6e23ddfe3a27d0cea0d80768de2d882bfe5bd5ca92f940b62a9fbfce42ef7
# containerA
# containerA
# test
# 
```

**Review by devimc — APPROVED** [review]
thanks @dgibson

**Pull request merged**